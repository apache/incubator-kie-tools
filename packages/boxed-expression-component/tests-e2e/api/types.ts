import {
  DMN15__tConditional,
  DMN15__tContext,
  DMN15__tDecisionTable,
  DMN15__tFilter,
  DMN15__tFor,
  DMN15__tFunctionDefinition,
  DMN15__tInvocation,
  DMN15__tList,
  DMN15__tLiteralExpression,
  DMN15__tQuantified,
  DMN15__tRelation,
} from "@kie-tools/dmn-marshaller/dist/schemas/dmn-1_5/ts-gen/types";

export type BoxedLiteral = { __$$element: "literalExpression" } & DMN15__tLiteralExpression;
export type BoxedRelation = { __$$element: "relation" } & DMN15__tRelation;
export type BoxedContext = { __$$element: "context" } & DMN15__tContext;
export type BoxedDecisionTable = { __$$element: "decisionTable" } & DMN15__tDecisionTable;
export type BoxedList = { __$$element: "list" } & DMN15__tList;
export type BoxedInvocation = { __$$element: "invocation" } & DMN15__tInvocation;
export type BoxedFunction = { __$$element: "functionDefinition" } & DMN15__tFunctionDefinition;
export type BoxedFor = { __$$element: "for" } & DMN15__tFor;
export type BoxedEvery = { __$$element: "every" } & DMN15__tQuantified;
export type BoxedSome = { __$$element: "some" } & DMN15__tQuantified;
export type BoxedConditional = { __$$element: "conditional" } & DMN15__tConditional;
export type BoxedFilter = { __$$element: "filter" } & DMN15__tFilter;

export type Normalized<T> = WithRequiredDeep<T, "@_id">;

type WithRequiredDeep<T, K extends keyof any> = T extends undefined
  ? T
  : T extends Array<infer U>
    ? Array<WithRequiredDeep<U, K>>
    : { [P in keyof T]: WithRequiredDeep<T[P], K> } & (K extends keyof T
        ? { [P in K]-?: NonNullable<WithRequiredDeep<T[P], K>> }
        : T);
