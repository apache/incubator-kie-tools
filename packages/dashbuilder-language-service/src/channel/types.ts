/*
 * Copyright 2022 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { TextDocument } from "vscode-languageserver-textdocument";
import { CompletionItemKind, Range } from "vscode-languageserver-types";

export declare type DashbuilderLsNodeType = "object" | "array" | "property" | "string" | "number" | "boolean" | "null";
export declare type DashbuilderJsonPath = (string | number)[];

/**
 * The AST node used in the LanguageServices
 */
export type DashbuilderLsNode = {
  type: DashbuilderLsNodeType;
  value?: any;
  offset: number;
  length: number;
  colonOffset?: number;
  parent?: DashbuilderLsNode;
  children?: DashbuilderLsNode[];
};

export interface TranslateArgs {
  completion: object | string;
  completionItemKind: CompletionItemKind;
  currentNodeRange?: Range;
  cursorOffset: number;
  document: TextDocument;
  overwriteRange?: Range;
}
