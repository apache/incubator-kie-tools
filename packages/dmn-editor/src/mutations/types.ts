import { NodeType } from "../diagram/connections/graphStructure";
import { NODE_TYPES } from "../diagram/nodes/NodeTypes";

export enum NodeNature {
  DRG_ELEMENT = "DRG_ELEMENT",
  ARTIFACT = "ARTIFACT",
}

export const nodeNatures: Record<NodeType, NodeNature> = {
  [NODE_TYPES.inputData]: NodeNature.DRG_ELEMENT,
  [NODE_TYPES.decision]: NodeNature.DRG_ELEMENT,
  [NODE_TYPES.bkm]: NodeNature.DRG_ELEMENT,
  [NODE_TYPES.knowledgeSource]: NodeNature.DRG_ELEMENT,
  [NODE_TYPES.decisionService]: NodeNature.DRG_ELEMENT,
  [NODE_TYPES.textAnnotation]: NodeNature.ARTIFACT,
  [NODE_TYPES.group]: NodeNature.ARTIFACT,
};
