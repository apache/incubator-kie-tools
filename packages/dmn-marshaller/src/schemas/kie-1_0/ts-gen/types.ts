// This file was automatically generated

import { XmlParserTsRootElementBaseType } from "@kie-tools/xml-parser-ts";

export type KIE__tComponentsWidthsExtension = {
  ComponentWidths?: KIE__tComponentWidths[]; // from type KIE__tComponentsWidthsExtension @ KIE.xsd
};

export type KIE__tComponentWidths = {
  "@_dmnElementRef"?: string; // from type KIE__tComponentWidths @ KIE.xsd
  width?: number[]; // from type KIE__tComponentWidths @ KIE.xsd
};

export type KIE__tAttachment = {
  "@_url"?: string; // from type KIE__tAttachment @ KIE.xsd
  "@_name"?: string; // from type KIE__tAttachment @ KIE.xsd
};
