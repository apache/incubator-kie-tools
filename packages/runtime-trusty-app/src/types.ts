/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
export interface TrustyContextValue {
  config: {
    counterfactualEnabled: boolean;
    explanationEnabled: boolean;
    serverRoot: string;
    basePath: string;
    useHrefLinks: boolean;
  };
}

export type RemoteData<E, D> =
  | { status: RemoteDataStatus.NOT_ASKED }
  | { status: RemoteDataStatus.LOADING }
  | { status: RemoteDataStatus.FAILURE; error: E }
  | { status: RemoteDataStatus.SUCCESS; data: D };

export enum RemoteDataStatus {
  NOT_ASKED,
  LOADING,
  FAILURE,
  SUCCESS,
}

export interface Execution {
  executionId: string;
  executionDate: string;
  executedModelName: string;
  executionType: string;
  executionSucceeded: boolean;
  executorName: string;
}

export interface Executions {
  headers: Execution[];
  total: number;
}

export interface ExecutionRouteParams {
  executionId: string;
  executionType: string;
}

export interface ItemObject {
  name: string;
  value: ItemObjectValue;
}

export type ItemObjectValue = ItemObjectUnit | ItemObjectCollection | ItemObjectStructure;

export interface ItemObjectUnit {
  kind: "UNIT";
  type: string;
  value: string | number | boolean | Array<string | number | boolean> | null;
}

export interface ItemObjectCollection {
  kind: "COLLECTION";
  type: string;
  value: ItemObjectValue[] | null;
}

export type ItemObjectMap = { [key: string]: ItemObjectValue };

export interface ItemObjectStructure {
  kind: "STRUCTURE";
  type: string;
  value: ItemObjectMap;
}

export interface InputRow {
  inputLabel: string;
  inputValue?: ItemObject["value"];
  key: string;
  category: string;
}

export enum evaluationStatus {
  EVALUATING = "Evaluating",
  FAILED = "Failed",
  NOT_EVALUATED = "Not evaluated",
  SKIPPED = "Skipped",
  SUCCEEDED = "Succeeded",
}

export type evaluationStatusStrings = keyof typeof evaluationStatus;

export interface Outcome {
  outcomeId: string;
  outcomeName: string;
  outcomeResult: ItemObjectValue;
  evaluationStatus: evaluationStatusStrings;
  hasErrors: boolean;
  messages: string[];
}

export interface FeatureScores {
  featureName: string;
  featureScore: number;
}

export interface Saliency {
  outcomeId: string;
  featureImportance: FeatureScores[];
}

export enum SaliencyStatus {
  SUCCEEDED = "SUCCEEDED",
  FAILED = "FAILED",
}

export type SaliencyStatusStrings = keyof typeof SaliencyStatus;

export interface Saliencies {
  status: SaliencyStatusStrings;
  statusDetail: string;
  saliencies: Saliency[];
}

export interface ServiceIdentifier {
  groupId?: string;
  artifactId?: string;
  version?: string;
}

export interface ModelData {
  deploymentDate?: string;
  modelId?: string;
  name: string;
  namespace: string;
  modelVersion: string;
  dmnVersion: string;
  serviceIdentifier: ServiceIdentifier;
  model: string;
}

export interface CFSearchInput {
  name: string;
  value: CFSearchInputValue;
}

export type CFSearchInputValue = CFSearchInputUnit | CFSearchInputCollection | CFSearchInputStructure;

export interface CFSearchInputUnit {
  kind: "UNIT";
  type: string;
  fixed?: boolean;
  domain?: CFNumericalDomain | CFCategoricalDomain;
  originalValue: ItemObject["value"];
}

export interface CFNumericalDomain {
  type: "RANGE";
  lowerBound?: number;
  upperBound?: number;
}

export interface CFCategoricalDomain {
  type: "CATEGORICAL";
  categories: string[];
}

export interface CFSearchInputCollection {
  kind: "COLLECTION";
  type: string;
  value: Array<CFSearchInputValue> | null;
}

export type CFSearchInputValueMap = { [key: string]: CFSearchInputValue };

export interface CFSearchInputStructure {
  kind: "STRUCTURE";
  type: string;
  value: CFSearchInputValueMap | null;
}

export enum CFGoalRole {
  UNSUPPORTED,
  ORIGINAL,
  FIXED,
  FLOATING,
}

export interface CFGoal {
  id: string;
  name: string;
  role: CFGoalRole;
  value: ItemObjectValue;
  originalValue: CFGoal["value"];
}

export type CFResult = Array<unknown>;

export interface CFStatus {
  isDisabled: boolean;
  executionStatus: CFExecutionStatus;
  lastExecutionTime: null | string;
}

export enum CFExecutionStatus {
  COMPLETED,
  RUNNING,
  NOT_STARTED,
  FAILED,
  NO_RESULTS,
}

export type CFAnalysisResetType = "NEW" | "EDIT";

export interface CFAnalysisExecution {
  executionId: string;
  counterfactualId: string;
  maxRunningTimeSeconds: number;
}

export interface CFAnalysisResult {
  executionId: string;
  counterfactualId: string;
  type: "counterfactual";
  valid: boolean;
  status: "SUCCEEDED" | "FAILED";
  statusDetails: string;
  solutionId: string;
  isValid: boolean;
  stage: "INTERMEDIATE" | "FINAL";
  inputs: ItemObject[];
  outputs: ItemObject[];
  sequenceId: number;
}

export interface CFAnalysisResultsSets extends CFAnalysisExecution {
  goals: CFGoal[];
  searchDomains: CFSearchInput[];
  solutions: CFAnalysisResult[];
}

export type CFSupportMessage = {
  id: string;
  message: string;
};
