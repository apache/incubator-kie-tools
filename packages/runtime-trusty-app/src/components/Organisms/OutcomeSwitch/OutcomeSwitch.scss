.outcome-switch {
  max-width: 400px;
}
