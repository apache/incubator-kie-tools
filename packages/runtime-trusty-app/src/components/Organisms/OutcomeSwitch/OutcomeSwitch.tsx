/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import React, { useEffect, useState } from "react";
import {
  Select,
  SelectDirection,
  SelectOption,
  SelectOptionObject,
  SelectVariant,
} from "@patternfly/react-core/dist/js/components";
import { Outcome } from "../../../types";
import "./OutcomeSwitch.scss";

type OutcomeSwitchProps = {
  currentExplanationId: string;
  onDecisionSelection: (outcomeId: string) => void;
  outcomesList: Outcome[];
};

const OutcomeSwitch = (props: OutcomeSwitchProps) => {
  const { outcomesList, onDecisionSelection, currentExplanationId } = props;
  const [isOpen, setIsOpen] = useState(false);
  const [selected, setSelected] = useState<string | SelectOptionObject>(currentExplanationId);

  const onToggle = (openStatus: boolean) => {
    setIsOpen(openStatus);
  };

  const onSelect = (event: React.MouseEvent | React.ChangeEvent, selection: string | SelectOptionObject) => {
    setSelected(selection);
    onDecisionSelection(selection as string);
    setIsOpen(false);
  };

  useEffect(() => {
    setSelected(currentExplanationId);
  }, [currentExplanationId]);

  return (
    <div className="outcome-switch">
      <Select
        variant={SelectVariant.single}
        aria-label="Select Decision Outcome"
        onToggle={onToggle}
        onSelect={onSelect}
        selections={selected}
        isOpen={isOpen}
        direction={SelectDirection.down}
      >
        {outcomesList.map((item, index) => (
          <SelectOption key={index} value={item.outcomeId}>
            {item.outcomeName}
          </SelectOption>
        ))}
      </Select>
    </div>
  );
};

export default OutcomeSwitch;
