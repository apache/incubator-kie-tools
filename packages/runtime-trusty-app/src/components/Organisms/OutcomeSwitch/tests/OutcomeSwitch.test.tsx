/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import React from "react";
import { mount, shallow } from "enzyme";
import { Outcome } from "../../../../types";
import OutcomeSwitch from "../OutcomeSwitch";

describe("ExplanationSwitch", () => {
  test("renders correctly", () => {
    const wrapper = shallow(<OutcomeSwitch {...props} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("SelectOption")).toHaveLength(2);
  });

  test("handles explanation selection", () => {
    const wrapper = mount(<OutcomeSwitch {...props} />);

    expect(wrapper.find("Select").prop("selections")).toMatch(props.currentExplanationId);

    wrapper.find("SelectToggle").simulate("click");

    wrapper.find('SelectOption[value="_9CFF8C35-4EB3-451E-874C-DB27A5A424C0"] button').simulate("click");

    expect(wrapper.find("Select").prop("selections")).toMatch("_9CFF8C35-4EB3-451E-874C-DB27A5A424C0");
    expect(props.onDecisionSelection).toHaveBeenCalledTimes(1);
    expect(props.onDecisionSelection).toHaveBeenCalledWith("_9CFF8C35-4EB3-451E-874C-DB27A5A424C0");
  });
});

const props = {
  outcomesList: [
    {
      outcomeId: "_12268B68-94A1-4960-B4C8-0B6071AFDE58",
      outcomeName: "Mortgage Approval",
      evaluationStatus: "SUCCEEDED",
      outcomeResult: {
        name: "Mortgage Approval",
        type: "boolean",
        value: true,
        components: [],
      },
      messages: [],
      hasErrors: false,
    },
    {
      outcomeId: "_9CFF8C35-4EB3-451E-874C-DB27A5A424C0",
      outcomeName: "Risk Score",
      evaluationStatus: "SUCCEEDED",
      outcomeResult: {
        name: "Risk Score",
        type: "number",
        value: 21.7031851958099,
        components: [],
      },
      messages: [],
      hasErrors: false,
    },
  ] as Outcome[],
  onDecisionSelection: jest.fn(),
  currentExplanationId: "_12268B68-94A1-4960-B4C8-0B6071AFDE58",
};
