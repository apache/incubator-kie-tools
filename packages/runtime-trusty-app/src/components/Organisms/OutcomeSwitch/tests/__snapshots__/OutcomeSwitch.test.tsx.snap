// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExplanationSwitch renders correctly 1`] = `
<div
  className="outcome-switch"
>
  <Select
    aria-describedby=""
    aria-invalid={false}
    aria-label="Select Decision Outcome"
    aria-labelledby=""
    chipGroupComponent={null}
    className=""
    clearSelectionsAriaLabel="Clear all"
    createText="Create"
    customBadgeText={null}
    customContent={null}
    direction="down"
    favorites={Array []}
    favoritesLabel="Favorites"
    hasInlineFilter={false}
    hasPlaceholderStyle={false}
    inlineFilterPlaceholderText={null}
    inputAutoComplete="off"
    inputIdPrefix=""
    isCreatable={false}
    isCreateOptionOnTop={false}
    isCreateSelectOptionObject={false}
    isDisabled={false}
    isFlipEnabled={true}
    isGrouped={false}
    isInputFilterPersisted={false}
    isInputValuePersisted={false}
    isOpen={false}
    isPlain={false}
    menuAppendTo="inline"
    noResultsFoundText="No results found"
    onClear={[Function]}
    onCreateOption={[Function]}
    onFilter={null}
    onSelect={[Function]}
    onToggle={[Function]}
    onTypeaheadInputChanged={null}
    ouiaSafe={true}
    placeholderText=""
    position="left"
    removeFindDomNode={false}
    removeSelectionAriaLabel="Remove"
    selections="_12268B68-94A1-4960-B4C8-0B6071AFDE58"
    shouldResetOnSelect={true}
    toggleAriaLabel="Options menu"
    toggleIcon={null}
    toggleId={null}
    toggleIndicator={null}
    typeAheadAriaDescribedby=""
    typeAheadAriaLabel=""
    validated="default"
    variant="single"
    width=""
    zIndex={9999}
  >
    <SelectOption
      className=""
      component="button"
      index={0}
      inputId=""
      isChecked={false}
      isDisabled={false}
      isFavorite={null}
      isLastOptionBeforeFooter={[Function]}
      isLoad={false}
      isLoading={false}
      isNoResultsOption={false}
      isPlaceholder={false}
      isSelected={false}
      key="0"
      keyHandler={[Function]}
      onClick={[Function]}
      sendRef={[Function]}
      setViewMoreNextIndex={[Function]}
      value="_12268B68-94A1-4960-B4C8-0B6071AFDE58"
    >
      Mortgage Approval
    </SelectOption>
    <SelectOption
      className=""
      component="button"
      index={0}
      inputId=""
      isChecked={false}
      isDisabled={false}
      isFavorite={null}
      isLastOptionBeforeFooter={[Function]}
      isLoad={false}
      isLoading={false}
      isNoResultsOption={false}
      isPlaceholder={false}
      isSelected={false}
      key="1"
      keyHandler={[Function]}
      onClick={[Function]}
      sendRef={[Function]}
      setViewMoreNextIndex={[Function]}
      value="_9CFF8C35-4EB3-451E-874C-DB27A5A424C0"
    >
      Risk Score
    </SelectOption>
  </Select>
</div>
`;
